### PR TITLE
#53 autofill contact number when sending sms invitation

### PR DESCRIPTION
**Pull request by Landry333**
#130 adding number auto fill when adding users

Setting auto fill number when a user want to add another user from phone book or by typing number and fixing/ removing the "send sms invitation" button from home page

Included UI test for SendSmsInvitationActivity

### CODE REVIEW

File: .idea/gradle.xml [hunk 1]
```diff
@@ -15,7 +15,6 @@
           </set>
```
Comment (refatK) [inline .idea/gradle.xml:15]: Please avoid pushing file changes for files from the idea folder next time.

**Comment by refatK** [comment]
Please put issue number in commit messages next time.

File: app/src/main/java/com/example/bigowlapp/activity/SearchContactsByPhone.java [hunk 2]
```diff
@@ -35,13 +30,16 @@
     private Button btnSearch;
     private final FirebaseFirestore db = FirebaseFirestore.getInstance();
     private static final ArrayList<QueryDocumentSnapshot> qds = new ArrayList<QueryDocumentSnapshot>();
+    private EditText inputNumber;
+    private String number2;
```
Comment (refatK) [inline app/src/main/java/com/example/bigowlapp/activity/SearchContactsByPhone.java:34]: You should use more descriptive name. I'm not sure what `number2` means. And `inputNumber` which I assume represent a phone number text input shot be like `inputPhoneNumber`

File: app/src/main/java/com/example/bigowlapp/activity/SearchContactsByPhone.java [hunk 3]
```diff
@@ -114,6 +114,7 @@ public void onComplete(@NonNull Task<QuerySnapshot> task) {
                                     else {
                                         Toast.makeText(SearchContactsByPhone.this, "User doesn't have the app", Toast.LENGTH_SHORT).show();
                                         Intent intent = new Intent(SearchContactsByPhone.this, SendSmsInvitationActivity.class);
+                                        intent.putExtra("number1", number2);
```
Comment (refatK) [inline app/src/main/java/com/example/bigowlapp/activity/SearchContactsByPhone.java:117]: same with intent extra ID, should be more descriptive.

File: app/src/androidTest/java/com/example/bigowlapp/activity/SendSmsInvitationActivityTest.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    public ActivityScenarioRule<SendSmsInvitationActivity> activityScenarioRule
+            = new ActivityScenarioRule<>(SendSmsInvitationActivity.class);
+
+    @Test
+    public void launchTextSmsOnButtonClick() {
+
+        onView(withId(R.id.number)).perform(typeText(STRING_TO_BE_TYPED),
+                closeSoftKeyboard());
+        onView(withId(R.id.send)).perform(click());
+
```
Comment (refatK) [inline app/src/androidTest/java/com/example/bigowlapp/activity/SendSmsInvitationActivityTest.java:35]: This test should assert something or Espresso check something.

Also maybe you should have a test that checks the autofill works if possible.

File: app/src/main/java/com/example/bigowlapp/activity/SendSmsInvitationActivity.java [hunk 5]
```diff
@@ -18,18 +19,38 @@
...
+        String contactDetails = getIntent().getStringExtra("number1");
+        String trimContactNumber = "000";
+        if (contactDetails != null) {
+            trimContactNumber = contactDetails.replaceAll("\\D+", "");
+        }
+        String contactNumber;
+        if(trimContactNumber.length()>=10){
+            contactNumber = trimContactNumber.substring(trimContactNumber.length() - 10);
+        }
+        else contactNumber = trimContactNumber;
```
Comment (refatK) [inline app/src/main/java/com/example/bigowlapp/activity/SendSmsInvitationActivity.java:42]: Could you explain what's going on here. Just curious.

Also, please fix style in this file. can just use `CTRL+ALT+SHIFT+L`

File: app/src/main/java/com/example/bigowlapp/activity/SendSmsInvitationActivity.java [hunk 6]
```diff
@@ -87,11 +98,9 @@ private void beginSendingProcess() {
         if (message2.isEmpty() && number2.isEmpty()) {
             Toast.makeText(SendSmsInvitationActivity.this, "Fields are empty!", Toast.LENGTH_SHORT).show();
         } else if (number2.isEmpty()) {
-            //Toast.makeText(SignUpPageActivity.this, "Please enter a phone number", Toast.LENGTH_SHORT).show();
```
Comment (refatK) [inline app/src/main/java/com/example/bigowlapp/activity/SendSmsInvitationActivity.java:90]: Thanks for removing commented code. 👍

**Review by refatK — CHANGES_REQUESTED** [review]
I have a couple of critcism and will also be trying to run the code.

I notices that `SearchContactsToSupervise.java` and `SearchContactsByPhone.java` have very similar and copy paste code (not from you). I'll make and issue to refactor this, so no need to worry about it now.

File: app/src/main/java/com/example/bigowlapp/activity/SendSmsInvitationActivity.java (hunk 5 above)
Comment (elvisno) [reply app/src/main/java/com/example/bigowlapp/activity/SendSmsInvitationActivity.java:42]: ok. I can fix style. This is to take the the last 10 digits of the input of the dialing  Edit text and use it as the text sms sending phone number

File: app/src/main/java/com/example/bigowlapp/activity/SendSmsInvitationActivity.java [hunk 7]
```diff
@@ -18,18 +19,38 @@
 public class SendSmsInvitationActivity extends AppCompatActivity {
 
     private EditText number, message;
-    private Button send, testSendRequest;
-    private String number2, message2;
+    private Button send;
+    private String number2, message2, noteText;
```
Comment (darkheros12) [inline app/src/main/java/com/example/bigowlapp/activity/SendSmsInvitationActivity.java:23]: I think you should put a more descriptive name for the variable. The name itself should explain what it does. If someone has to refactor the code or debug, they should not spend too much time figuring out what the variable does.

File: app/src/main/java/com/example/bigowlapp/activity/SendSmsInvitationActivity.java [hunk 8]
```diff
@@ -18,18 +19,38 @@
...
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_send_sms_invitation);
 
-        number = findViewById(R.id.number);
+        noteTv = (TextView) findViewById(R.id.note);
+
+        String contactDetails = getIntent().getStringExtra("number1");
```
Comment (darkheros12) [inline app/src/main/java/com/example/bigowlapp/activity/SendSmsInvitationActivity.java:33]: By fixing the above, the number1 should become more descriptive.

**Review by darkheros12 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/bigowlapp/activity/SearchContactsByPhone.java (hunk 3 above)
Comment (Landry333) [reply app/src/main/java/com/example/bigowlapp/activity/SearchContactsByPhone.java]: ok. Changes have been done

File: app/src/main/java/com/example/bigowlapp/activity/SendSmsInvitationActivity.java (hunk 7 above)
Comment (Landry333) [reply app/src/main/java/com/example/bigowlapp/activity/SendSmsInvitationActivity.java]: ok. I have renamed the variables

File: app/src/main/java/com/example/bigowlapp/activity/SendSmsInvitationActivity.java (hunk 8 above)
Comment (Landry333) [reply app/src/main/java/com/example/bigowlapp/activity/SendSmsInvitationActivity.java]: I has been fixed

**Review by darkheros12 — COMMENTED** [review]
The naming is good for me. It is more descriptive.

**Review by darkheros12 — APPROVED** [review]

File: .idea/gradle.xml (hunk 1 above)
Comment (Landry333) [reply .idea/gradle.xml:15]: ok. Probably didn't see that

File: app/src/androidTest/java/com/example/bigowlapp/activity/SendSmsInvitationActivityTest.java (hunk 4 above)
Comment (Landry333) [reply app/src/androidTest/java/com/example/bigowlapp/activity/SendSmsInvitationActivityTest.java:35]: ok. Will look into that

File: app/src/main/java/com/example/bigowlapp/activity/SearchContactsByPhone.java (hunk 3 above)
Comment (refatK) [reply app/src/main/java/com/example/bigowlapp/activity/SearchContactsByPhone.java]: thanks

**Review by refatK — COMMENTED** [review]
I ran the app and have a couple of questions

**Comment by refatK** [comment]
{

Are these suppose to happen?

**Comment by refatK** [comment]
2 1 Screenshot 2020-11-08 142003
When I load smsinvitation on its own, its says the contact is null, and starts with `000`. Looks weird.

Also when I click back from this page, it seems like it tries to go back to Loginpage instead of HomePage, so it looks a bit wonky.

**Comment by Landry333** [comment]
The format is 10 digit maximum like numbers in Canada. So didn't include +1 as all numbers are supposed to be in Canada. This is from the way the data is produced. It comes as the contact name attached with the number. For the other UI problem I will make an update

**Review by refatK — APPROVED** [review]
Seems fine now. Thanks for addressing all my feedback

**Pull request merged**